diff --git a/src/state/shellscripts/bwrap.sh b/src/state/shellscripts/bwrap.sh
index 6f5d7dbe..56f0cfbc 100755
--- a/src/state/shellscripts/bwrap.sh
+++ b/src/state/shellscripts/bwrap.sh
@@ -11,7 +11,11 @@ fi
 
 ARGS=(--unshare-net --new-session)
 ARGS=("${ARGS[@]}" --proc /proc --dev /dev)
-ARGS=("${ARGS[@]}" --bind /tmp /tmp --tmpfs /run --tmpfs /var)
+ARGS=("${ARGS[@]}" --bind "$TMPDIR" /tmp \
+                   --setenv TMPDIR /tmp --setenv TMP /tmp --setenv TEMPDIR /tmp --setenv TEMP /tmp \
+                   --ro-bind "$(command -v sh)" /bin/sh \
+                   --ro-bind "$(command -v bash)" /bin/bash \
+                   --tmpfs /run --tmpfs /var)
 
 add_mounts() {
     case "$1" in
@@ -23,7 +27,7 @@ add_mounts() {
     done
 }
 
-add_mounts ro /usr /bin /lib /lib32 /lib64 /etc /opt /nix/store /rw/usrlocal /home
+add_mounts ro /usr /lib /lib32 /lib64 /etc /opt /nix/store /rw/usrlocal /home
 
 # C compilers using `ccache` will write to a shared cache directory
 # that remain writeable. ccache seems widespread in some Fedora systems.
